app/api/orders.py

```diff
@@ -284,15 +284,27 @@ def before_update_object(self, order, data, view_kwargs):
             if current_user.id == order.user_id:
                 # Order created from the tickets tab.
                 for element in data:
-                    if data[element] and data[element]\
+                    if element not in ['event', 'ticket_holders', 'user'] and data[element] and data[element]\
                             != getattr(order, element, None) and element not in get_updatable_fields():
                         raise ForbiddenException({'pointer': 'data/{}'.format(element)},
                                                  "You cannot update {} of an order".format(element))
+                    elif element in ['event', 'user'] and data[element] and data[element]\
+                            != str(getattr(order, element, None).id) and element not in get_updatable_fields():
+                        raise ForbiddenException({'pointer': 'data/{}'.format(element)},
+                                                 "You cannot update {} of an order".format(element))
+                    elif element == 'ticket_holders' and data[element]:
+                        ticket_holders = []
+                        for ticket_holder in order.ticket_holders:
+                            ticket_holders.append(str(ticket_holder.id))
+                        if data[element] != ticket_holders and element not in get_updatable_fields():
+                            raise ForbiddenException({'pointer': 'data/{}'.format(element)},
+                                                     "You cannot update {} of an order".format(element))
 
             else:
                 # Order created from the public pages.
                 for element in data:
-                    if data[element] and data[element] != getattr(order, element, None):
+                    if element not in ['event', 'ticket_holders', 'user'] and data[element] and data[element]\
```

Comment:
`data.get(elemen) != order.get(element)`

Reply:
data[element] will throw an exception if the key doesn't exist

Reply:
@poush actually we are looping over the elements which are in `data` itself (on line 305), that's why I didn't follow the approach you mentioned.

Reply:
I didn't get If you are already iterating over the `data` dict and what is the point of writing `data[element] `?

Reply:
If the element is in data, and here in data[element] you checking for the null or false value of it then you can simply put a parent `if` block to check that. This will prevent un-necessary execution of the statements below.

for eg: 
``` 
if data[element]:
    if .... :
    elif .. :
    elif ... :

Reply:
or simply to prevent if-else ladder, you can put
``` 
if not data[element]:
    continue
```